cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc

```diff
@@ -269,12 +269,21 @@ bool SubstraitToVeloxPlanValidator::validateCast(
   const auto& toType = SubstraitParser::parseType(castExpr.type());
   core::TypedExprPtr input = exprConverter_->toVeloxExpr(castExpr.input(), inputType);
 
-  // Only support cast from date to timestamp
-  if (toType->kind() == TypeKind::TIMESTAMP && !input->type()->isDate()) {
-    LOG_VALIDATION_MSG(
-        "Casting from " + input->type()->toString() + " to " + toType->toString() + " is not supported.");
-    return false;
-  }
+  // Only support cast from date or integral type to timestamp.
+  if (toType->kind() == TypeKind::TIMESTAMP)
+    if (!input->type()->isDate()) {
```

Comment:
We can still use one `if` here, same as the original one `if (toType->kind() == TypeKind::TIMESTAMP && !input->type()->isDate())`